src/linker/Linker/TypeReferenceExtensions.cs

```diff
@@ -358,6 +359,12 @@ public static bool IsTypeOf<T> (this TypeReference tr)
 			return tr.Name == type.Name && tr.Namespace == tr.Namespace;
 		}
 
+		public static bool IsTypeOf (this TypeReference tr, WellKnownType type)
+		{
+			var (@namespace, name) = type.GetNamespaceAndName ();
+			return tr.IsTypeOf (@namespace, name);
```

Comment:
Cecil has a similar concept to SpecialType - it's called MetadataType and through it one can detect primitive types (and couple of other cases) faster than string compares.

For example https://github.com/dotnet/linker/blob/ae1f280e2f58d0007a4a0acbb254b06e17da9b98/src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs#L505 does a boolean check.

Reply:
Good to know, I'll add that in a new PR. Thanks!